spec/controllers/comments_controller_spec.rb

```diff
@@ -226,6 +220,22 @@
                                      "access the page you were trying to " \
                                      "reach. Please log in.")
         end
+
+        context "when the commentable is a restricted work" do
+          let(:work) { comment.ultimate_parent }
+
+          before { work.update!(restricted: true) }
+
+          it "(on a comment) redirects to the login page" do
```

Comment:
Can I be annoying for a sec? If I'm reading this correctly, I don't really love this new test being under `context "when the commentable is a restricted work"`. It seems like the commentable is actually a _comment_, while the restricted work is the ultimate parent.

So maybe it's more like: 
```
when the ultimate parent is a restricted work
  when the commentable is also the work
    it redirects to the login page
  when the commentable is a comment
    it redirects to the login page
```
Or:
```
when the commentable is a restricted work
  it redirects to the login page
when the commentable is a comment on a restricted work
  it redirects to the login page
```